lib/common/strings.c

```diff
@@ -898,24 +898,21 @@ pcmk__str_in_list(GList *lst, const gchar *s, uint32_t flags)
 }
 
 static bool
-str_any_of(bool casei, const char *s, va_list args)
+str_any_of(const char *s, va_list args, uint32_t flags)
 {
-    bool rc = false;
-
     if (s != NULL) {
         while (1) {
             const char *ele = va_arg(args, const char *);
 
             if (ele == NULL) {
                 break;
-            } else if (pcmk__str_eq(s, ele,
-                                    casei? pcmk__str_casei : pcmk__str_none)) {
-                rc = true;
-                break;
+            } else if (pcmk__str_eq(s, ele, flags)) {
+                return true;
             }
         }
     }
-    return rc;
+
+    return false;
```

Comment:
I wonder if we should honor null_matches when s==NULL ...

Reply:
That would be easy to do, but there's currently no way for str_any_of to get flags aside from pcmk__str_casei.  It's wrapped by pcmk__strcase_any_of and pcmk__str_any_of, so those functions would have to grow the ability to accept flags first.

Reply:
True, but I think it makes sense to put the infrastructure in str_any_of() for future-proofing, since it's now accepting generic flags

Reply:
Added.